apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -48,11 +55,59 @@ const InnerFileBrowser = React.memo(
     moveFilePrompt,
     renameFilePrompt,
     renameFolderPrompt,
-    toggleFileVisibility,
+    setFileVisibility,
+    setFileIsValidation,
   }: FilesComponentProps) => {
     const {openFile, deleteFile, toggleOpenFolder, deleteFolder} =
       useCodebridgeContext();
     const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+    const hasValidationFile = Object.values(files).some(f => f.validation);
+
+    const handleSetFileVisibility = (fileId: string, hidden: boolean) => {
+      setFileVisibility(fileId, hidden);
+      // When we set the visibility, we also want to set the file to validation: false.
+      setFileIsValidation(fileId, false);
+    };
```

Comment:
This is saying that we always set validation to false if we're changing the visibility. i.e., if we set visibility to true or false, we always set validation to false.

Is this correct behavior, or should it only be setting validation to false for one of the visibility cases?

Reply:
This is the correct behavior, although this comment and the one below by @mikeharv are making me think of doing a bit of a refactor here.

> How well understood do you feel like the 3-way toggle between visible/hidden/validation is by our internal users?

In the model I had, if the validation boolean was `true`, the hidden boolean is ignored (the file is always hidden). However, the terminology for hidden/visible in Java Lab is "support" and "starter". 

To Mike's question, this functionally copies the behavior in Java Lab, which is the same internal user. In Java Lab things are implemented a bit differently, because validation code goes into a separate, encrypted json blob,  while the start/support files are in the same start code blob. I made all files in the same json here because we decided we didn't need to encrypt the validation. In Java Lab we also use different terminology; validation is the same, but visible files are called "starter" and hidden are called "support". I am thinking it would be good to use the same terminology here as well. Looking at these comments I will also change the two booleans (`hidden` and `validation`) into one "file type" field that can be one of `"starter" | "support" | "validation"`. This will make it clearer a file is only one of the three options.